test/parallel/test-fs-readdir-pipe.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const { readdir, readdirSync } = require('fs');
+
+if (!common.isWindows) {
+  common.skip('This test is specific to Windows to test enumerate pipes');
+}
+
+// Ref: https://github.com/nodejs/node/issues/56002
+// This test is specific to Windows.
+
+const pipe = '\\\\.\\pipe\\';
+
+assert.ok(readdirSync(pipe).length >= 0);
+
+readdir(pipe, common.mustCall((err, files) => {
+  assert.ok(err == null);
+  assert.ok(files.length >= 0);
+}));
```

Comment:
Let's use `mustSucceed` to get more informative errors
```suggestion
readdir(pipe, common.mustSucceed((files) => {
  assert.ok(files.length >= 0, `${files.length} is not greater or equal to 0`);
}));
```